src/internal/wishlist.js

```diff
@@ -1,139 +1,106 @@
-    uploadSync(cfg);
+export function setWishlist(wishlist, wishlistId) {
+  switch (wishlistId) {
+    case 'wish':
+      localStorageSet(CONSTS.wish, JSON.stringify(wishlist));
+      break;
+    case 'trade':
+      localStorageSet(CONSTS.trade, JSON.stringify(wishlist));
+      break;
+    default:
+      break;
   }
 }
 
-export function getWishlist() {
-  const w2 = localStorageLoad(CONSTS.wishlistV2);
-  if (w2) {
+export function getWishlist(wishlistId) {
+  const ws = localStorageLoad(CONSTS[wishlistId]);
+  if (ws) {
     try {
-      return JSON.parse(w2);
+      return JSON.parse(ws);
     } catch (e) {
       console.log('Unable to read the Wishlist v2 object');
     }
   }
   const d = getDefaultWishlist();
-  setWishlist(d);
+  setWishlist(d, wishlistId);
   return d;
 }
 
-export function addCap(id) {
-  const w = getWishlist();
-  w.items.push({ id, prio: false });
-  setWishlist(w);
-  return w;
-}
-
-export function rmCap(id) {
-  const w = getWishlist();
-  w.items = w.items.filter((x) => x.id !== id);
-  setWishlist(w);
-  return w;
-}
-
-export function isInWishlist(w, id) {
-  return w && w.items && w.items.findIndex((x) => x.id === id) > -1;
-}
-
-export function addTradeCap(id) {
-  const w = getWishlist();
-  w.tradeItems.push({ id, prio: false });
-  setWishlist(w);
-  return w;
-}
-
-export function rmTradeCap(id) {
-  const w = getWishlist();
-  w.tradeItems = w.tradeItems.filter((x) => x.id !== id);
-  setWishlist(w);
-  return w;
-}
-
-export function isInTradeList(w, id) {
-  return w && w.tradeItems && w.tradeItems.findIndex((x) => x.id === id) > -1;
-}
-
-export async function uploadSync(cfg) {
-  const wishlist = getWishlist();
-  if (cfg.wishlist_id) {
-    updateCollection(cfg.wishlist_id, {
-      name: CONSTS.wishlistV2,
-      wishlist,
-    });
-  } else {
-    const id = await setCollection({
-      name: CONSTS.wishlistV2,
-      wishlist,
-    });
-
-    const config = { ...cfg, wishlist_id: id };
-
-    setConfig(config);
-  }
+export function getLocalCollections() {
+  return [
+    {
+      content: getWishlist('wish'),
+      name: 'Wishlist',
+      id: 'wish',
+    },
+    {
+      content: getWishlist('trade'),
+      name: 'Tradelist',
+      id: 'trade',
+    },
+  ];
 }
 
-export async function downloadSync(cfg) {
-  const collections = await getCollections();
-  if (collections.length) {
-    const config = { ...cfg, wishlist_id: collections[0].id };
-
-    console.log(collections);
-    localStorageSet(CONSTS.wishlistV2, JSON.stringify(collections[0].content));
-    setConfig(config);
+export function initMigrateWishlist() {
+  let ws = localStorageLoad(CONSTS.wishlistV2);
+  if (ws) {
+    try {
+      ws = JSON.parse(ws);
+      localStorageSet(CONSTS.wish, JSON.stringify({ items: ws.items }));
+      localStorageSet(CONSTS.trade, JSON.stringify({ items: ws.tradeItems }));
+      localStorageSet(CONSTS.settings, JSON.stringify(ws.settings));
```

Comment:
at the end of the migration we need to delete the wishlistv2, otherwise it will endlessly migrate everytime you load the page.

Reply:
you're right, i have fixed